xapian-core/api/vectortermlist.h

```diff
@@ -66,14 +70,33 @@ class VectorTermList : public TermList {
 	}
 	data.reserve(total_size);
 
+	std::string prev;
 	// Now encode all the terms into data.
 	for (I i = begin; i != end; ++i) {
-	    pack_string(data, *i);
+	    if (rare(prev.size() > 255))
+		prev.resize(255);
+	    const std::string& term = *i;
+	    size_t reuse = common_prefix_length(prev, term);
+	    data.append(1, char(reuse));
+	    pack_uint(data, term.size() - reuse);
+	    data.append(term, reuse, std::string::npos);
+	    prev = term;
 	}
```

Comment:
This makes the encoding for existing uses of `VectorTermList` more verbose, and also increases the cost of encoding - I'm not sold that's a good trade-off.  There are inherently differences between the sorted and unsorted cases which are highlighted by the changes you've had to make here.

Can't we just use `RemoteMetaDataTermList` for this?  That seems to handle essentially the same situation of a sorted `TermList` from a remote database.

Reply:
Yes, Since now RemoteMetaDataTermList isn't just handling only metadata but also synonyms and probably spellings in future when it's implemented.

Would it make sense to change the name from RemoteMetaDataTermList to something which depicts it's just a list of sorted terms (without term frequency and num_entries )?
maybe RemoteSortedTermList

Reply:
> Would it make sense to change the name from RemoteMetaDataTermList to something which depicts it's just a list of sorted terms (without term frequency and num_entries )?
maybe RemoteSortedTermList

I think we should rename, but let's do it as a separate step so the functional changes here can be seen without the noise of the renaming.

I'm also not sure what the best name is - `RemoteSortedTermList` is confusing as we already have `RemoteTermList` which is also a sorted list of terms.  Indeed almost all `TermList` subclasses are sorted (I think `VectorTermList` is the only one which isn't) and none of the others have `Sorted` in the name.

Probably the key difference compared with `RemoteTermList` is that `RemoteMetaDataTermList` is just terms without any frequency information.